src/equal_str.cpp

```diff
@@ -41,28 +43,27 @@ bool EqualStr::evaluate(ov::TensorVector& outputs, const ov::TensorVector& input
     outputs[0].set_shape(ov::Shape{ num_elems });
     auto result = outputs[0].data<int32_t>();
 
-    for (size_t idx = 0; idx < num_elems; ++idx) {
+    ov::parallel_for(num_elems, [&](size_t idx){
         // handle indices due to broadcasting case
-        size_t idx1 = (idx < num_elems1) ? idx : 0;
-        size_t idx2 = (idx < num_elems2) ? idx : 0;
-        auto begin1 = begins1[idx1];
-        auto begin2 = begins2[idx2];
-        auto end1 = ends1[idx1];
-        auto end2 = ends2[idx2];
+        const size_t idx1 = (idx < num_elems1) ? idx : 0;
+        const size_t idx2 = (idx < num_elems2) ? idx : 0;
+        const auto begin1 = begins1[idx1];
+        const auto begin2 = begins2[idx2];
+        const auto end1 = ends1[idx1];
+        const auto end2 = ends2[idx2];
 
         if ((end1 - begin1) != (end2 - begin2)) {
             // unequal string lengths guarantees false result
             result[idx] = 0;
-            continue;
         } else if (end1 == begin1) {
             // both empty strings case
             result[idx] = 1;
-            continue;
-        }
+        };
```

Comment:
let us remove this if-else construction:
1. auto chars1 = inputs[2].data<const char>();
2. std::string str1(chars1 + begin1, chars1 + end1);
3. result[idx] = (str1 == str2 ? 1 : 0);